### PR TITLE
Port more ConcurrentSnapshotsIT cases

### PR DESCRIPTION
**Pull request by mfussenegger**
Follow up to https://github.com/crate/crate/pull/13478

## TODO

- [x] testMultipleReposAreIndependent
- [x] testMultipleReposAreIndependent2
- [x] testMultipleReposAreIndependent3
- [x] testSnapshotRunsAfterInProgressDelete
- [x] testAbortOneOfMultipleSnapshots

Will follow up with the remainder in more PRs:

- [ ] testCascadedAborts
- [ ] testMasterFailOverWithQueuedDeletes
- [ ] testAssertMultipleSnapshotsAndPrimaryFailOver
- [ ] testQueuedDeletesWithFailures
- [ ] testQueuedDeletesWithOverlap
- [ ] testQueuedOperationsOnMasterRestart
- [ ] testQueuedOperationsOnMasterDisconnect
- [ ] testQueuedOperationsOnMasterDisconnectAndRepoFailure
- [ ] testQueuedOperationsAndBrokenRepoOnMasterFailOver
- [ ] testQueuedSnapshotOperationsAndBrokenRepoOnMasterFailOver
- [ ] testQueuedSnapshotOperationsAndBrokenRepoOnMasterFailOver2
- [ ] testQueuedSnapshotOperationsAndBrokenRepoOnMasterFailOverMultipleRepos
- [ ] testMultipleSnapshotsQueuedAfterDelete
- [ ] testQueuedSnapshotsWaitingForShardReady
- [ ] testBackToBackQueuedDeletes
- [ ] testQueuedOperationsAfterFinalizationFailure
- [ ] testStartDeleteDuringFinalizationCleanup
- [ ] testEquivalentDeletesAreDeduplicated
- [ ] testMasterFailoverOnFinalizationLoop
- [ ] testStatusMultipleSnapshotsMultipleRepos
- [ ] testInterleavedAcrossMultipleRepos
- [ ] testMasterFailoverAndMultipleQueuedUpSnapshotsAcrossTwoRepos
- [ ] testConcurrentOperationsLimit

### CODE REVIEW

**Comment by matriv** [comment]
Just a tip: maybe add `@Repeat(iterations=50)` or so while it's in progress, as some tests where also flaky.

File: server/src/testFixtures/java/io/crate/testing/SQLResponseAssert.java [hunk 1]
```diff
@@ -39,6 +39,13 @@ public SQLResponseAssert hasRowCount(long expectedRowCount) {
         return this;
     }
 
+    public SQLResponseAssert hasRows(String ... rows) {
+        String result = TestingHelpers.printedTable(actual.rows());
+        String[] resultRows = result.split("\n");
+        assertThat(resultRows).contains(rows);
```
Comment (matriv) [inline server/src/testFixtures/java/io/crate/testing/SQLResponseAssert.java:45]: `containsExactly()`?

File: server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java [hunk 2]
```diff
@@ -21,29 +21,58 @@
...
 import io.crate.integrationtests.disruption.discovery.AbstractDisruptionTestCase;
 import io.crate.testing.UseRandomizedSchema;
 
 @UseRandomizedSchema(random = false)
 @IntegTestCase.ClusterScope(scope = IntegTestCase.Scope.TEST, numDataNodes = 0)
 public class ConcurrentSnapshotsIT extends AbstractSnapshotIntegTestCase {
 
+    // Large snapshot pool settings to set up nodes for tests involving multiple repositories that need to have enough
+    // threads so that blocking some threads on one repository doesn't block other repositories from doing work
+    private static final Settings LARGE_SNAPSHOT_POOL_SETTINGS = Settings.builder()
```
Comment (matriv) [inline server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java:71]: Could that be the solution to the problems we were facing? I hope so!

**Review by matriv — APPROVED** [review]
LGTM, left a comment regarding the assertion, thx!

File: server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java:71]: This was also part of the original PR, so probably not.

**Pull request merged**